Add cpuMax & memoryReserved properties

yuxiaojian opened the pull request:
The old doc doesn't have cpuMax/memoryMax properties. The default value cpuMax/memoryMax is 500m/700Mi. This is ok if customers special smaller values for cpuReserved/memoryReserved but the with greater values than 500m/700Mi, the deployment fails.

yuxiaojian committed:
Add cpuMax & memoryReserved properties

The old doc doesn't have cpuMax/memoryMax properties. The default value cpuMax/memoryMax is 500m/700Mi. This is ok if customers special smaller values for cpuReserved/memoryReserved but the with greater values than 500m/700Mi, the deployment fails.

Commit:
Minor style updates

paigehargrave reviewed (changes requested):
Please take a quick look at my updates to ensure I did not introduce any errors. Also, can someone verify that `500m` is correct on line 72 & 161 (looks like it should be vCores),  and`800Mi` is correct on line 74 & 146 (looks like it should be MB), Thank you!

Commit:
Removed cpumax and memorymax info

paigehargrave commented:
Updates made per Sundip's feedback.

sundippatel commented on modules/ROOT/pages/deploy-maven-3.x.adoc:
```diff
@@ -1,20 +1,27 @@
+and working with Maven plugins.
 
-Before deploying an application to Runtime Fabric you should understand the number of resource it requires. See xref:runtime-fabric::deploy-resource-allocation.adoc[Resource Allocation on Anypoint Runtime Fabric] for more information.
+Before deploying an application to Runtime Fabric you should understand the number of resource it requires. 
```
resource**s**

paigehargrave replied:
Got it - thank you!

sundippatel commented on modules/ROOT/pages/deploy-maven-3.x.adoc:
```diff
@@ -1,20 +1,27 @@
 
-To deploy a Mule application using Maven, you must add the Mule Maven Plugin to the application project. Add the plugin to you project by adding the following dependency:
+To deploy a Mule application using Maven, you must add the Mule Maven Plugin to the application project. 
+Add the plugin to you project by adding the following dependency:
```
your

paigehargrave replied:
Got it - thank you!

sundippatel commented on modules/ROOT/pages/deploy-maven-3.x.adoc:
```diff
@@ -81,28 +88,29 @@ This example uses placeholder values. Configure each value with values specific
 == Redeploy to Runtime Fabric
 
-To redeploy the application, run the same command as you did to deploy. Runtime Fabric rewrites the application you had deployed.
+To redeploy the application, run the same command as you did to deploy. Runtime Fabric rewrites the 
```
"as you did to deploy" -- is there better wording we could use there? 

"as you ran to deploy" is better, IMO.

sundippatel replied:
the word "rewrites" is interesting.. all it's doing is re-deploying with the new artifact (the application).

paigehargrave replied:
Reworded this.

sundippatel commented on modules/ROOT/pages/deploy-maven-3.x.adoc:
```diff
@@ -81,28 +88,29 @@ This example uses placeholder values. Configure each value with values specific
 If not set, the default value is `+https://anypoint.mulesoft.com+`.
-| muleVersion | specifies the Mule runtime version installed in your Runtime Fabric instance. +
+| muleVersion | Specifies the Mule runtime version installed in your Runtime Fabric instance. +
 Ensure that this value is at equal or higher than the minimum required Mule runtime version of your application.
```
consider removing "at"

paigehargrave replied:
Got it - thank you!

sundippatel commented on modules/ROOT/pages/deploy-maven-3.x.adoc:
```diff
@@ -81,28 +88,29 @@ This example uses placeholder values. Configure each value with values specific
-| applicationName | specifies the name of your application deployed in Exchange. +
+| username | Specifies your Anypoint Platform username.
+| password | Specifies your Anypoint Platform password.
+| applicationName | Specifies the name of your application deployed in Exchange. +
```
Exchange is where we store artifacts (like applications). Instead of "deployed", perhaps we update that to "stored".

Then we can reserve the "deploy" term for Runtime Fabric. (In context, when deploying an application to Runtime Fabric using the Maven plugin, the application is stored in Exchange as a prerequisite, implicit step.)

paigehargrave replied:
Got it - thank you!

sundippatel commented on modules/ROOT/pages/deploy-maven-3.x.adoc:
```diff
@@ -81,28 +88,29 @@ This example uses placeholder values. Configure each value with values specific
+| username | Specifies your Anypoint Platform username.
+| password | Specifies your Anypoint Platform password.
+| applicationName | Specifies the name of your application deployed in Exchange. +
 This name is part of the domain of your deployed app.
```
"Depending on the configuration of your internal load balancer, this name may be used as a subdomain used to access your application."

paigehargrave replied:
Added this.

sundippatel commented on modules/ROOT/pages/deploy-maven-3.x.adoc:
```diff
@@ -81,28 +88,29 @@ This example uses placeholder values. Configure each value with values specific
+| applicationName | Specifies the name of your application deployed in Exchange. +
 This name is part of the domain of your deployed app.
-| businessGroup (optional) | specifies the business group where you want to deploy the application. If not defined, name of the master organization is used as the default.
+| businessGroup (optional) | Specifies the business group where you want to deploy the application. If not defined, the name of the master organization is used as the default.
```
I think we can simplify a bit: "If not defined, the master organization is.."

paigehargrave replied:
Got it - thank you!

sundippatel commented on modules/ROOT/pages/deploy-maven-3.x.adoc:
```diff
@@ -81,28 +88,29 @@ This example uses placeholder values. Configure each value with values specific
-| provider | specifies the rovider type. The only supported value for Runtime Fabric is `MC`.
-| environment | specifies the Anypoint Platform environment where you want to deploy your application.
-| properties (optional) | specifies a top-level element. +
+| target | Specifies the Runtime Fabric target name where the applicateion is deployed.
```
typo on application

paigehargrave replied:
Got it - thank you!

sundippatel commented on modules/ROOT/pages/deploy-maven-3.x.adoc:
```diff
@@ -81,28 +88,29 @@ This example uses placeholder values. Configure each value with values specific
-| environment | specifies the Anypoint Platform environment where you want to deploy your application.
-| properties (optional) | specifies a top-level element. +
+| target | Specifies the Runtime Fabric target name where the applicateion is deployed.
+| provider | Specifies the rovider type. The only supported value for Runtime Fabric is `MC`.
```
typo on provider

sundippatel replied:
The wording here is weird. We could say, "Specifies the provider type; for Runtime Fabric, set this value to 'MC'".

paigehargrave replied:
Got 'em. Thank you!

sundippatel commented on modules/ROOT/pages/deploy-maven-3.x.adoc:
```diff
@@ -81,28 +88,29 @@ This example uses placeholder values. Configure each value with values specific
-| properties (optional) | specifies a top-level element. +
+| target | Specifies the Runtime Fabric target name where the applicateion is deployed.
+| provider | Specifies the rovider type. The only supported value for Runtime Fabric is `MC`.
+| environment |Specifies the Anypoint Platform environment where you want to deploy your application.
```
Consider adding a space after the pipe.

paigehargrave replied:
Done.

sundippatel commented on modules/ROOT/pages/deploy-maven-3.x.adoc:
```diff
@@ -116,32 +124,35 @@ For example:
+The default value is `false`.
+| clusteringEnabled (optional) | Enables Mule clustering across each replica of the application. 
+You must have at least two replicas of your application. The default value is `false`.
+| memoryReserved (optional) | Defines the amount of memory allocated for each replica of the application. 
```
"..of the application." -> "of your application." for consistency w/ rest of the page.

paigehargrave replied:
Got it - thanks!

sundippatel commented on modules/ROOT/pages/deploy-maven-3.x.adoc:
```diff
@@ -116,32 +124,35 @@ For example:
 
-This will set 0.5 vCores for each replica.
-| server (optional) | specifies the Maven server that contains Anypoint Platform credentials. This is property is only required if you want to use the credentials stored in your Maven `settings.xml` file. Note: This is not the Mule server name.
+| server (optional) | Specifies the Maven server that contains Anypoint Platform credentials. This is property is only required if you want to use the credentials stored in your Maven `settings.xml` file. Note: This is not the Mule server name.
```
Repetitive "is" -- "This property is.."

paigehargrave replied:
Got it - thank you!

sundippatel reviewed (changes requested):
Changes are good. Left some comments and suggestions - all minor.

Commit:
Updates per Sundip

Commit:
Minor syntax updates

sarathecoubian commented on modules/ROOT/pages/deploy-maven-3.x.adoc:
```diff
@@ -1,20 +1,27 @@
-To deploy an application to Runtime Fabric, the application must be published in Exchange. You can use Studio, as describe in  xref:studio::export-to-exchange-task.adoc[Publishing a Project to Exchange]. Alternatively, use the Mule Maven Plugin. See xref:exchange::to-publish-assets-maven.adoc[Publish and Deploy Exchange Assets Using Maven] for more information.
-
+To deploy an application to Runtime Fabric, the application must be published in Exchange. 
+You can use Studio, as describe in  xref:studio::export-to-exchange-task.adoc[Publishing a Project to Exchange]. 
```
"describe" to "described"

paigehargrave replied:
Got it - thanks!

sarathecoubian commented on modules/ROOT/pages/deploy-maven-3.x.adoc:
```diff
@@ -1,20 +1,27 @@
+and working with Maven plugins.
 
-Before deploying an application to Runtime Fabric you should understand the number of resource it requires. See xref:runtime-fabric::deploy-resource-allocation.adoc[Resource Allocation on Anypoint Runtime Fabric] for more information.
+Before deploying an application to Runtime Fabric, you should understand the number of resources it requires. 
```
Is "it" from "number of resources it requires" referring to the application, or Runtime Fabric?

paigehargrave replied:
Fixed.

sarathecoubian commented on modules/ROOT/pages/deploy-maven-3.x.adoc:
```diff
@@ -81,67 +88,69 @@ This example uses placeholder values. Configure each value with values specific
-This name is part of the domain of your deployed app.
-| businessGroup (optional) | specifies the business group where you want to deploy the application. If not defined, name of the master organization is used as the default.
+| muleVersion | Specifies the Mule runtime version installed in your Runtime Fabric instance. +
+Ensure that this value is equal to or later than the minimum required Mule runtime version of your application.
```
"Equal to" is a bit confusing in this context.

paigehargrave replied:
Fixed - thank you!

sarathecoubian commented on modules/ROOT/pages/deploy-maven-3.x.adoc:
```diff
@@ -81,67 +88,69 @@ This example uses placeholder values. Configure each value with values specific
-| environment | specifies the Anypoint Platform environment where you want to deploy your application.
-| properties (optional) | specifies a top-level element. +
+| target | Specifies the Runtime Fabric target name where the application is deployed.
+| provider | Specifies the provider type. for Runtime Fabric, set this value to `MC`.
```
Capitalize "for".

paigehargrave replied:
Got it - thanks!

sarathecoubian commented on modules/ROOT/pages/deploy-maven-3.x.adoc:
```diff
@@ -81,67 +88,69 @@ This example uses placeholder values. Configure each value with values specific
+Ensure that this value is equal to or later than the minimum required Mule runtime version of your application.
+| username | Specifies your Anypoint Platform username.
+| password | Specifies your Anypoint Platform password.
+| applicationName | Specifies the name of your application stored in Exchange. +
```
application that is stored in Exchange.

paigehargrave replied:
Got it - thanks!

sarathecoubian reviewed (approved):
Looks good!

Commit:
Updates per Sara's review

paigehargrave commented:
@sundippatel  - ready for final approval.

sundippatel committed:
Minor edit

Removed a repetitive sentence that's outdated after adding in suggested changes.

sundippatel reviewed (approved):
Looks great!
